llvm/lib/Target/AMDGPU/VOP3PInstructions.td

```diff
@@ -480,10 +480,12 @@ multiclass VOP3PDOTF8Inst <string OpName, SDPatternOperator intrinsic_node> {
                                           i32:$src2_modifiers, f32:$src2)>;
 }
 
+let OtherPredicates = [isGFX12Plus] in {
```

Comment:
Why not [HasDot9Insts]?

Reply:
It looks to me these dot4_f32_* are only generated with gfx1200. And I don't know exactly whether they belong to Dot9Insts. If yes, we may use [isGFX12Plus, HasDot9Insts]